docker: create key for customized user and fix expected output in TestSingleNodeDocker

ZhouXing19 opened the pull request:
The recent `single-node-secure-mode` test in the CI is failing with
`open certs/client.myuser.crt: no such file or directory` error.
This commit is to fix it by creating key for the customized user name.

This commit also updated expected output due to the move to pgx.

fixes #82349

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/82354)
<!-- Reviewable:end -->

knz reviewed (approved):
thank you!

ZhouXing19 commented:
Note to myself:
With the changes, the test still failed with 
```
--- FAIL: TestSingleNodeDocker (11.96s)
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_355_355)        --- FAIL: TestSingleNodeDocker/single-node-insecure-mode (1.51s)
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_356_356)            single_node_docker_test.go:223: cannot run fsnotify to listen to init_success:
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_357_357)                res:&docker.execResult{stdOut:"", stdErr:"", exitCode:137}: command "./docker-fsnotify /cockroach init_success 80" exit with code 137:
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_358_358)                 {stdOut: stdErr: exitCode:137}
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_359_359)        --- FAIL: TestSingleNodeDocker/single-node-insecure-mem-mode (1.26s)
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_360_360)            single_node_docker_test.go:223: cannot run fsnotify to listen to init_success:
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_361_361)                res:&docker.execResult{stdOut:"", stdErr:"", exitCode:137}: command "./docker-fsnotify /cockroach init_success 80" exit with code 137:
[17:23:16 ](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_ScratchProjectPutTcExperimentsInHere_JaneDockerImageTest/5355085?showRootCauses=true&expandBuildChangesSection=true&expandBuildDeploymentsSection=true&expandBuildProblemsSection=true&showLog=5355085_362_362)                 {stdOut: stdErr: exitCode:137}
```
Seems that it's because the container for `single-node-insecure-mode` didn't successfully spin up.

ZhouXing19 committed:
docker: create key for customized user and fix expected output in TestSingleNodeDocker

The recent single-node-secure-mode test in the CI is failing with
open certs/client.myuser.crt: no such file or directory error.
This commit is to fix it by creating key for the customized user name.

This commit also updated expected output due to the move to pgx.

fixes #82349

Release note: None

<what was there before: Previously, ...>
<why it needed to change: This was inadequate because ...>
<what you did about it: To address this, this patch ...>

Release note (<category, see below>): <what> <show> <why>

ZhouXing19 commented:
Thanks!
bors r=knz
